File: tests/conformance/fixtures/typescript/01-basic.ts
import { Pipeline } from '../../../../sdk/typescript/src/index';

const p = new Pipeline();

p.task('lint').run('npm run lint').inputs('**/*.ts');

p.task('test').run('npm test').after('lint')
  .inputs('**/*.ts', '**/*.test.ts').timeout(120);

p.task('build').run('npm run build').after('test');

p.emit();

File: tests/conformance/fixtures/go/01-basic.go
package main

import sykli "github.com/yairfalse/sykli/sdk/go"

func main() {
	p := sykli.New()

	p.Task("lint").Run("npm run lint").Inputs("**/*.ts")

	p.Task("test").Run("npm test").After("lint").
		Inputs("**/*.ts", "**/*.test.ts").Timeout(120)

	p.Task("build").Run("npm run build").After("test")

	p.Emit()
}
